From 74142849a42682aa68c9bfbcd2953363479842f7 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Thu, 9 Jan 2020 14:51:44 +0100
Subject: [PATCH] PONRTSYS-5630: Limit number of MAC addresses learned per port
 by default

Restrict the number of MAC addresses which can be learned per bridge
port by default to the maximum possible number (254) instead of
unlimited which is the hardware default. The userspace can change this
later to a different number or unlimited again.
---
 drivers/net/datapath/dpm/gswip31/datapath_switchdev.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/datapath/dpm/gswip31/datapath_switchdev.c b/drivers/net/datapath/dpm/gswip31/datapath_switchdev.c
index d7f567585f3a..ff705f200278 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_switchdev.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_switchdev.c
@@ -108,7 +108,14 @@ int dp_swdev_bridge_port_cfg_set(struct br_info *br_item,
 	brportcfg.nBridgeId = br_item->fid;
 	brportcfg.nBridgePortId = bport;
 	brportcfg.eMask = GSW_BRIDGE_PORT_CONFIG_MASK_BRIDGE_ID |
-			  GSW_BRIDGE_PORT_CONFIG_MASK_BRIDGE_PORT_MAP;
+			  GSW_BRIDGE_PORT_CONFIG_MASK_BRIDGE_PORT_MAP |
+			  GSW_BRIDGE_PORT_CONFIG_MASK_MAC_LEARNING_LIMIT;
+
+	/* Limit the MAC learning on each port by default to 254 entries.
+	 * The PRX300 SDL TMR requires us to limit this by default.
+	 */
+	brportcfg.nMacLearningLimit = GSWIP_LEARN_LIMIT_PORT_MAX;
+	brportcfg.bMacLearningLimitEnable = 1;
 	ret = gsw_bp->BridgePort_ConfigSet(gsw_handle, &brportcfg);
 	if (ret != GSW_statusOk) {
 		pr_err("Fail in allocating/configuring bridge port\n");
